### PR TITLE
Add an 'info' prop in input to display info-styled icon and message

### PR DESCRIPTION
**Pull request by taoliusage**
# Description
`InputValidation`: now accepts a `info` prop to display info-styled icon and message attached to an input.

# TODO
- [x] Release notes

# Related Issues / Pull Requests
https://github.com/Sage/sage_one_advanced/pull/12325

# Screenshots / Gifs
screen shot 2017-06-20 at 6 10 33 pm 2
screen shot 2017-06-20 at 7 23 02 pm 2

### CODE REVIEW

**Comment by abramin** [comment]
It looks like the arrow has disappeared?

File: src/utils/decorators/input/input.scss [hunk 1]
```diff
@@ -290,3 +251,35 @@
   top: 6px;
   right: -24px;
 }
+
+@mixin common-input-theme($type, $color, $border, $background, $color-hover) {
```
Comment (toni-leigh) [inline src/utils/decorators/input/input.scss]: `$border`, `$background` and `$color-hover` don't seem to be used in this mixin

Comment (taoliusage) [inline src/utils/decorators/input/input.scss]: Addressed.

**Comment by taoliusage** [comment]
@benwilsonsage Would you like to have a styled warning beside the input field (e.g. info styled, helper styled, etc. as shown above) which provides more flexibility on how we want to present warning for the input field? Or, do we just want to add an info styled warning now (for the email input field)? Andrew was worried that with styled warning the icon and theme may not match the validation summary beside the Save button. Please let us know your thoughts. Thanks.

**Comment by benwilsonsage** [comment]
Hi @taoliusage ! Sorry for the delay coming back to you. I've caught up with the threat, and the approach suggested by Andrew sounds best - separate to errors/warnings, so it doesn't show up in the form summary. Thanks for your help!

**Comment by taoliusage** [comment]
@andrewjtait I have re-implemented the design based on the feedback from you and Ben.

File: src/utils/decorators/input-validation/input-validation.js [hunk 2]
```diff
@@ -518,9 +591,15 @@ let InputValidation = (ComposedComponent) => class Component extends ComposedCom
...
    */
   get validationHTML() {
-    if (this.state.valid && !this.state.warning) { return null; }
+    let type = '';
+
+    if (this.state.valid && !this.state.warning && !this.state.info) { return null; }
 
-    let type = !this.state.valid ? "error" : "warning";
+    if (this.state.info) {
+      type = "info";
```
Comment (andrewjtait) [inline src/utils/decorators/input-validation/input-validation.js]: Just to confirm, we are stating that the order of importance is `info` then `error` then `warning`? May need to confirm this with UX.

File: src/utils/decorators/input-validation/input-validation.js [hunk 3]
```diff
@@ -537,7 +616,7 @@ let InputValidation = (ComposedComponent) => class Component extends ComposedCom
       <Icon key="0" ref="validationIcon" type={ type } className={ iconClasses } style={ iconStyle } />,
       <div key="1" className="common-input__message-wrapper" onMouseOver={ this.showMessage } onMouseOut={ this.hideMessage }>
         <div ref="validationMessage" className={ messageClasses }>
-          { this.state.errorMessage || this.state.warningMessage }
+          { this.state.errorMessage || this.state.warningMessage || this.state.infoMessage }
```
Comment (andrewjtait) [inline src/utils/decorators/input-validation/input-validation.js]: however this order says `error` then `warning` then `info`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+# 1.1.3
```
Comment (andrewjtait) [inline CHANGELOG.md]: will need to target at least `1.2.0` (maybe even `1.3.0` as I think `1.2.0` will be released soon)

**Comment by andrewjtait** [comment]
Spoke with UX - order of importance should be `error` -> `warning` -> `info`. So if we have an input with all 3, and all 3 are triggered we should display `error`. Once that is resolved display the `warning`, and finally display the `info`.

**Comment by taoliusage** [comment]
@benwilsonsage Could you please take a look at the screenshots in the PR description to see if the improvement looks good? Thanks.

**Comment by benwilsonsage** [comment]
Looks great to me @taoliusage ! Thanks for your help.

**Comment by abramin** [comment]
@taoliusage Looks ok, checked against existing behaviour in GAC contacts dialog.

**Pull request merged**